sympy/matrices/common.py

```diff
@@ -2434,37 +2434,43 @@ def pow(self, exp, method=None):
             if exp < 0:
                 exp = -exp
                 a = a.inv()
-            # When certain conditions are met,
-            # Jordan block algorithm is faster than
-            # computation by recursion.
-            elif method == 'jordan':
-                try:
-                    return jordan_pow(exp)
-                except MatrixError:
-                    if method == 'jordan':
-                        raise
-
-            if method == 'cayley':
-                return a._eval_pow_by_cayley(exp)
+        # When certain conditions are met,
+        # Jordan block algorithm is faster than
+        # computation by recursion.
+        if method == 'jordan':
+            try:
+                return jordan_pow(exp)
+            except MatrixError:
+                if method == 'jordan':
+                    raise
 
-            elif method == "mulsimp":
-                return a._eval_pow_by_recursion_dotprodsimp(exp)
+        elif method == 'cayley':
+            if not exp.is_Number or exp % 1 != 0:
+                raise ValueError("cayley method is only valid for integer powers")
```

Comment:
nonnegative integer powers?

Reply:
when the  power is negative, then the matrix is `inverted` and `exp *= -1` is done in previous step. 
So, even  method = `multiply` works for  negative powers.